src/bug_report.rs

```diff
@@ -0,0 +1,198 @@
+use crate::utils::exec_cmd;
+use std::fs;
+use std::path::PathBuf;
+
+pub fn create() {
+    let os_info = os_info::get();
+
+    let environment = Environment {
+        os_type: os_info.os_type(),
+        os_version: os_info.version().to_owned(),
+        shell_info: get_shell_info(),
+        starship_config: get_starship_config(),
+    };
+
+    let link = make_github_issue_link(crate_version!(), environment);
+
+    if open::that(&link)
+        .map(|exit_status| exit_status.success())
+        .unwrap_or(false)
+    {
+        println!(
+            "I was unable to launch your browser. You'll have to copy this link instead:\n\n{}",
+            link
+        );
+    }
+}
+
+const UNKNOWN_SHELL: &str = "<unknown shell>";
+const UNKNOWN_VERSION: &str = "<unknown version>";
+const UNKNOWN_CONFIG: &str = "<unknown config>";
+
+struct Environment {
+    os_type: os_info::Type,
+    os_version: os_info::Version,
+    shell_info: ShellInfo,
+    starship_config: String,
+}
+
+fn make_github_issue_link(starship_version: &str, environment: Environment) -> String {
+    let title = urlencoding::encode("Bug Report:");
+
+    let body = urlencoding::encode(&format!("## Bug Report
```

Comment:
Given how many parameters there are, I think it would be a good idea to use [named parameters](https://doc.rust-lang.org/std/fmt/#named-parameters) to make future adjustments easier.